src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/BrokeredServices/ServiceBrokerFactory.cs

```diff
@@ -30,13 +32,16 @@ internal class ServiceBrokerFactory
     private readonly ExportProvider _exportProvider;
     private Task _bridgeCompletionTask;
     private readonly CancellationTokenSource _cancellationTokenSource = new CancellationTokenSource();
+    private readonly ImmutableArray<IOnServiceBrokerInitialized> _onServiceBrokerInitialized;
 
     [ImportingConstructor]
     [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public ServiceBrokerFactory(ExportProvider exportProvider)
+    public ServiceBrokerFactory([ImportMany] IEnumerable<IOnServiceBrokerInitialized> onServiceBrokerInitialized,
```

Comment:
This will activate all the MEF parts immediately. Do you want to do that, or lazily activate them in time to call the one method on the interface?

Reply:
As it is, it's fine because the LSP service is guaranteed to be created if coming from an LSP factory.  It is 99.99% guaranteed when the broker service is initialized, the LSP service is already loaded.  For that 0.01 % case, the TCS will take care of awaiting for synchronization.